google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/SpannerServiceContributor.java

```diff
@@ -43,10 +48,21 @@
   @Override
   public void contribute(StandardServiceRegistryBuilder serviceRegistryBuilder) {
     serviceRegistryBuilder
+        // The custom Hibernate schema management tool for Spanner.
+        .addInitiator(new StandardServiceInitiator() {
+          @Override
+          public Service initiateService(Map configurationValues,
+              ServiceRegistryImplementor registry) {
+            return SCHEMA_MANAGEMENT_TOOL;
+          }
+
+          @Override
+          public Class getServiceInitiated() {
+            return SchemaManagementTool.class;
+          }
+        })
         // The user agent JDBC connection property to identify the library.
         .applySetting("hibernate.connection.userAgent", HIBERNATE_API_CLIENT_LIB_TOKEN)
-        // The custom Hibernate schema management tool for Spanner.
-        .applySetting("hibernate.schema_management_tool", SCHEMA_MANAGEMENT_TOOL)
```

Comment:
So, was this an incorrect way to specify the schema management tool, and initiator should be use instead?
What about the other `.applySetting()` calls we have? Do those also not work on Quarkus?
Will the `userAgent` be set?

Reply:
> So, was this an incorrect way to specify the schema management tool, and initiator should be use instead?

Yeah, Sanne claimed setting properties is not expected to be done in the service contributor. The applySetting didn't work for changing the schema management tool in Quarkus and so I had to rely on the initiator method.

> What about the other .applySetting() calls we have? Do those also not work on Quarkus?
> Will the userAgent be set?

Good question, I will have to dig deeper into this. Right now I am not sure.

Reply:
Filed #184 to track.